[draft] caching website page

wiardvanrij opened the pull request:
Relates to mainly #3888 

and somewhat #3669 

This is an initial PR. I would love more people involved to make the text accurate, solid and useful.

wiardvanrij commented on docs/caching.md:
```diff
@@ -0,0 +1,238 @@
+
+## Query frontend specific for in-memory
+
+`validity` specifies cache valid time , If set to 0s, so using a default of 24 hours expiration time
```
I'm actually not sure if this is correct, as I could not find information on this :D

wiardvanrij commented on docs/caching.md:
```diff
@@ -0,0 +1,238 @@
+However it is possible to give guidance on what each component tries to achieve, what it is used for and how you could configure this.
+
+<sub>
+1. service level agreement<br />
```
I would love to remove the 'brs' but somehow markdown totally ruins the style for new lines by enters.

GiedriusS replied:
Weird :thinking: was about to comment about this

wiardvanrij replied:
It adds a 1.5 newline between the 2nd and 3rd line. Not sure how to fix it 🥺

wiardvanrij commented on docs/caching.md:
```diff
@@ -0,0 +1,238 @@
+
+## Bucket cache specific for in-memory
+
+`in-memory` bucket cache is not supported. This is only possible with `memcached` as backend.
```
I'm somewhat confused, as the current docs live on the website state that it is supported here: https://thanos.io/tip/components/store.md/#caching-bucket 

Yet I also saw a place were it states that only memcached is supported.

GiedriusS replied:
Maybe we forgot to update something? https://github.com/thanos-io/thanos/pull/3579

GiedriusS commented on docs/caching.md:
```diff
@@ -0,0 +1,238 @@
+
+## Query frontend cache
+
+Query Frontend supports caching query results and reuses them on subsequent queries. If the cached results are incomplete, Query Frontend calculates the required subqueries and executes them in parallel on downstream queriers. Query Frontend can optionally align queries with their step parameter to improve the cacheability of the query results.
```
Query frontend also caches label requests :P

GiedriusS commented on docs/caching.md:
```diff
@@ -0,0 +1,238 @@
+
+## Query frontend specific
+
+`expiration` specifies memcached cache valid time , If set to 0s, so using a default of 24 hours expiration time
```
`expiration` means how long the data is stored in Memcached i.e. after that time Memcached evicts those items. Yep, the default is `24h` :hugs: I think we can improve this sentence:
```suggestion
`expiration` specifies how long memcached itself keeps items. After that time, memcached evicts those items. `0s` means the default duration of `24h`.
```

GiedriusS commented on docs/caching.md:
```diff
@@ -0,0 +1,238 @@
+
+## Index cache specific for in-memory
+
+The index cache has no exceptions on the base configuration
```
Full stop missing

GiedriusS reviewed (commented):
Maybe also worth mentioning that different services can share the same memcached instance(-s), for example? I see such questions regularly on Slack. Thank you for your work :hugs:

yashrsharma44 commented on docs/caching.md:
```diff
@@ -0,0 +1,238 @@
+However it is possible to give guidance on what each component tries to achieve, what it is used for and how you could configure this.
+
+<sub>
+1. service level agreement<br />
+2. service level objective<br />
+3. service level indicator
```
```suggestion
- service level agreement
- service level objective
- service level indicator
```

wiardvanrij commented on docs/caching.md:
```diff
@@ -0,0 +1,238 @@
+
+# Background
+
+Thanos can implement caching layers on multiple components. In general this should speed-up certain processes. There is no definitive guide on how one should setup an entire Thanos stack as each cluster and use-case is different. Especially what KPI's<sup>1</sup>, SLO's<sup>2</sup> and SLI's<sup>3</sup> matter to you, your customers and/or your stack should have influence on the setup.
```
Yes, totally agree. I would like more input on this though as I lack the 100% correct knowledge on that subject.

wiardvanrij commented:
For easier reference a screenshot of the table of contents: 
![image](https://user-images.githubusercontent.com/5786097/111036378-3b610a00-841f-11eb-9b0c-1ff3d247e147.png)

wiardvanrij commented:
Recent update removed specific configuration information from component pages as I explained in the issue #3888 
IMO this makes each component easier to understand and specific caching configuration can now be found on the 'caching' page. 

I understand this might look weird and not logical but imagine the use-case: A new person that has no affinity with Thanos yet. The component page can be quite complicated. By moving specific information and guiding the user to the correct part, it should become more clear.

Please also checkout this branch and run it locally `make web-serve` to check it out. 

For easier reference i've included a screenshot of the table of contents of the caching page.  I'm 100% open for any feedback on this change in #3888 . It's just merely a suggestion, so don't worry about giving feedback. 


![image](https://user-images.githubusercontent.com/5786097/111036589-3486c700-8420-11eb-82b1-cfb86e09358e.png)

wiardvanrij commented:
> Maybe also worth mentioning that different services can share the same memcached instance(-s), for example? I see such questions regularly on Slack. Thank you for your work 🤗

I've added this, but I could use extra help on wording because I added the following note:

```
It however is not possible to use one memcached instance for multiple store backends as this provides conflicts on index keys.```

I've tested this myself with a Thanos setup that consists of two different setups having a different backend. What will happen if they both use the same memcached instance, is that it will fetch indexes from the index cache (?) and tries to fetch this from the storage backend. Yet it does not exists as it's not part of that setup.

wiardvanrij commented:
I'm going to close this. I need to refactor the page and include all new changes included. Will make a new PR today!
